File: backend/app/api/v1/configs.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.core.database import get_db
from app.api.deps import get_current_user
from app.schemas.config import ConfigCreate, ConfigUpdate
from app.services.config.config_service import ConfigService

router = APIRouter(prefix="/configs", tags=["配置"])


def _ok(data=None, message="success"):
    return {"code": 200, "message": message, "data": data}


@router.get("/")
async def list_configs(
    current_user: dict = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    service = ConfigService(db)
    items = service.get_all_configs()
    return _ok({"items": items})


@router.get("/{key}")
async def get_config(
    key: str,
    current_user: dict = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    service = ConfigService(db)
    config = service.get_config(key)
    if config is None:
        return {"code": 404, "message": f"配置 '{key}' 不存在", "data": None}
    return _ok(config)


@router.post("/")
async def create_config(
    config_data: ConfigCreate,
    current_user: dict = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    service = ConfigService(db)
    result = service.set_config(
        key=config_data.key,
        value=config_data.value,
        value_type=config_data.value_type,
        description=config_data.description,
        is_sensitive=config_data.is_sensitive,
    )
    return _ok(result)


@router.put("/{key}")
async def update_config(
    key: str,
    config_data: ConfigUpdate,
    current_user: dict = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    service = ConfigService(db)
    result = service.set_config(
        key=key,
        value=config_data.value,
        value_type=config_data.value_type,
        description=config_data.description,
    )
    return _ok(result)


@router.delete("/{key}")
async def delete_config(
    key: str,
    current_user: dict = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    service = ConfigService(db)
    deleted = service.delete_config(key)
    if not deleted:
        return {"code": 404, "message": f"配置 '{key}' 不存在", "data": None}
    return _ok(message="删除成功")

File: services/api/src/api/v1/configs.rs
use axum::Router;
use axum::routing::get;
use axum::extract::{State, Path};
use axum::Json;
use serde::Deserialize;
use serde_json::{json, Value};
use std::sync::Arc;

use crate::services::config::ConfigService;

#[derive(Clone)]
pub struct ConfigState {
    pub config_service: Arc<ConfigService>,
}

#[derive(Deserialize)]
pub struct CreateConfigRequest {
    pub key: String,
    pub value: String,
    pub value_type: Option<String>,
    pub description: Option<String>,
    pub is_sensitive: Option<bool>,
}

#[derive(Deserialize)]
pub struct UpdateConfigRequest {
    pub value: Option<String>,
    pub value_type: Option<String>,
    pub description: Option<String>,
    pub is_sensitive: Option<bool>,
}

pub async fn list_configs(State(state): State<ConfigState>) -> Json<Value> {
    match state.config_service.list().await {
        Ok(items) => { let total = items.len(); Json(json!({"code": 200, "data": {"items": items, "total": total}})) }
        Err(e) => Json(json!({"code": 500, "message": e.to_string()})),
    }
}

pub async fn get_config(State(state): State<ConfigState>, Path(key): Path<String>) -> Json<Value> {
    match state.config_service.get(&key).await {
        Ok(Some(item)) => Json(json!({"code": 200, "data": item})),
        Ok(None) => Json(json!({"code": 404, "message": "配置项不存在"})),
        Err(e) => Json(json!({"code": 500, "message": e.to_string()})),
    }
}

pub async fn create_config(State(state): State<ConfigState>, Json(req): Json<CreateConfigRequest>) -> Json<Value> {
    match state.config_service.create(
        &req.key, &req.value, req.value_type.as_deref(),
        req.description.as_deref(), req.is_sensitive,
    ).await {
        Ok(item) => Json(json!({"code": 200, "data": item})),
        Err(e) => Json(json!({"code": 400, "message": e.to_string()})),
    }
}

pub async fn update_config(State(state): State<ConfigState>, Path(key): Path<String>, Json(req): Json<UpdateConfigRequest>) -> Json<Value> {
    match state.config_service.update(
        &key, req.value.as_deref(), req.value_type.as_deref(),
        req.description.as_deref(), req.is_sensitive,
    ).await {
        Ok(item) => Json(json!({"code": 200, "data": item})),
        Err(e) => Json(json!({"code": 400, "message": e.to_string()})),
    }
}

pub async fn delete_config(State(state): State<ConfigState>, Path(key): Path<String>) -> Json<Value> {
    match state.config_service.delete(&key).await {
        Ok(()) => Json(json!({"code": 200, "message": "删除成功"})),
        Err(e) => Json(json!({"code": 400, "message": e.to_string()})),
    }
}

pub fn routes(state: ConfigState) -> Router {
    Router::new()
        .route("/configs", get(list_configs).post(create_config))
        .route("/configs/{key}", get(get_config).put(update_config).delete(delete_config))
        .with_state(state)
}
